File: tests/fixtures/go/clean_comment_length.go
package main

// FetchWithRetry pings the upstream health endpoint a few times before giving up, retrying
// internally so callers never have to implement their own retry loop around this same flaky
// dependency that intermittently fails for a moment right after each fresh deploy completes.
func FetchWithRetry() bool {
	// Retries a few times before giving up; the upstream flakes briefly after each deploy.
	for i := 0; i < 3; i++ {
		if upstream() {
			return true
		}
	}
	return false
}

func upstream() bool {
	return true
}

File: tests/fixtures/python/clean_comment_length.py
def fetch_with_retry():
    # Retries a few times before giving up; the upstream flakes briefly after each deploy.
    for _ in range(3):
        if upstream():
            return True
    return False


def upstream():
    """Ping the upstream health endpoint and report whether it responded successfully this
    time, retrying internally a small number of times before giving up and returning False so
    callers never have to implement their own retry loop around this same flaky dependency.
    """
    return True
